torch/_export/utils/__init__.py

```diff
@@ -0,0 +1,105 @@
+import copy
+import dataclasses
+import weakref
+import re
+from collections import OrderedDict
+from typing import Any, Callable, List, Tuple, Optional, Dict, Union
+
+import sympy
+
+import torch
+import torch._dynamo
+import torch.fx
+import torch.utils._pytree as pytree
+import torch._export
+
+from torch.fx.graph import _PyTreeCodeGen, _PyTreeInfo
+
+def _unlift(gm, inp_pos_to_buffer_name, in_spec, out_spec, state_dict):
+    count = 0
+    # Step 1: make lifted params as get_attr
+    for node in gm.graph.nodes:
+        if node.op == "placeholder":
+            if count in inp_pos_to_buffer_name:
+                with gm.graph.inserting_after(node):
+                    getattr_node = gm.graph.get_attr(inp_pos_to_buffer_name[count])
+                    node.replace_all_uses_with(getattr_node)
+                    gm.graph.erase_node(node)
+            count += 1
+
+    # Step 2: Fix the input/output of the graph now that we deleted
+    # some args.
+    gm.graph.lint()
+    names = [f"arg_{i}" for i in range(len(in_spec.children_specs))]
+    gm.graph._codegen = _PyTreeCodeGen(
+        _PyTreeInfo(
+            names,
+            in_spec,
+            out_spec,
+        )
+    )
+    gm.recompile()
+
+    # Step 3: Find state references in HigherOrderOps and recursively
+    # fix them.
```

Comment:
Does this work for nested conds too?

Reply:
Yeah it should. I will add a test. My test covers nested map + cond, so nested cond should be similar.